From 74d01a9e6d19eb963bbf081e3ea7c692759a8fe3 Mon Sep 17 00:00:00 2001
From: Stefan Ghinea <stefan.ghinea@windriver.com>
Date: Mon, 18 Dec 2023 01:20:21 +0000
Subject: [PATCH] Find krb5-config for cross-compilation

With $dir being used as base search for both libgssapi and krb5-config,
in OE krb5-config is found in $dir/bin/crossscripts instead of $dir/bin
so add this to the search posibillities relative to $dir.
Remove check for needing to set -rpath in KRBLDFLAGS.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Stefan Ghinea <stefan.ghinea@windriver.com>
Signed-off-by: Haitao Liu <haitao.liu@windriver.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 aclocal/kerberos5.m4 | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/aclocal/kerberos5.m4 b/aclocal/kerberos5.m4
index f96f0fd..61bc81e 100644
--- a/aclocal/kerberos5.m4
+++ b/aclocal/kerberos5.m4
@@ -26,6 +26,8 @@ AC_DEFUN([AC_KERBEROS_V5],[
       K5CONFIG=$dir/bin/krb5-config
     elif test -f "/usr/kerberos/bin/krb5-config"; then
       K5CONFIG="/usr/kerberos/bin/krb5-config"
+    elif test -f $dir/bin/crossscripts/krb5-config; then
+      K5CONFIG=$dir/bin/crossscripts/krb5-config
     elif test -f "/usr/lib/mit/bin/krb5-config"; then
       K5CONFIG="/usr/lib/mit/bin/krb5-config"
     fi
@@ -71,17 +73,6 @@ AC_DEFUN([AC_KERBEROS_V5],[
   fi
   AC_MSG_RESULT($KRBDIR)
 
-  dnl Check if -rpath=$(KRBDIR)/lib is needed
-  echo "The current KRBDIR is $KRBDIR"
-  if test "$KRBDIR/lib" = "/lib" -o "$KRBDIR/lib" = "/usr/lib" \
-       -o "$KRBDIR/lib" = "//lib" -o "$KRBDIR/lib" = "/usr//lib" ; then
-    KRBLDFLAGS="";
-  elif /sbin/ldconfig -p | grep > /dev/null "=> $KRBDIR/lib/"; then
-    KRBLDFLAGS="";
-  else
-    KRBLDFLAGS="-Wl,-rpath=$KRBDIR/lib"
-  fi
-
   dnl Now check for functions within gssapi library
   AC_CHECK_LIB($gssapi_lib, gss_krb5_export_lucid_sec_context,
     AC_DEFINE(HAVE_LUCID_CONTEXT_SUPPORT, 1, [Define this if the Kerberos GSS library supports gss_krb5_export_lucid_sec_context]), ,$KRBLIBS)
-- 
2.25.1

